scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua

```diff
@@ -43,268 +104,84 @@ end;
-    -- Obviously, you have to move the Avatars to their own family, and give them access to AFlows via a new set of moves.
-    -- Might be able to cheat by giving them a copy AFlow (change the name!) that despawns the mob once completed.
-    -- Rearranging the skins may be necessary to use this trick efficiently on more SMNs.
-    -- Either that, or probably somewhat complex core code.  Avatars may not always be mobid+1.
-    -- It wasn't clear if the avatars were a separate pop, or if all dead lamps should revive, go avatar, and AFlow.
-
-    --[[
-    ------------------------ Astral Flow Logic ------------------------
-    -- Missing the log message for players.  Needs to be handled in the core somehow.
-    -- Possibly supposed to use twice per trigger?  Did not check too far on this.  Sounds fun.
-    if (mobHP <= (mobMaxHP * 0.75) and target:getMaskBit(PWardenAstralFlows,3) == false) then
-        for i=17056178, 17056186 do
-            local rannum = math.random(0,7);
-            local avatar = GetMobByID(i);
-            avatar:changeSkin(23 + rannum); -- Random avatar skin
-            SpawnMob(i):updateEnmity(target);
-            avatar:useMobAbility(912);
-            DespawnMob(i);
-        end
-        PWardenAstralFlows = PWardenAstralFlows + 4;
-        -- 23 = Shiva, 628 Diamond Dust
-        -- 24 = Ramuh, 637 Judgement Bolt
-        -- 25 = Titan, 601 Earthen Fury
-        -- 26 = Ifrit, 592 Inferno
-        -- 27 = Leviathan, 610 Tidal Wave
-        -- 28 = Garuda, 619 Aerial Blast
-        -- 29 = Fenrir, 583 Howling Moon
-        -- 30 = Carbuncle, 656 Searing Light
-        -- 31 = Diabolos
-        -- 646 = wyvern breath.  Need to find diabolos.
-
-    elseif (mobHP <= (mobMaxHP * 0.5) and target:getMaskBit(PWardenAstralFlows,2) == false) then
-        for i=17056178, 17056186 do
-            local rannum = math.random(0,7);
-            local avatar = GetMobByID(i);
-            avatar:changeSkin(23 + rannum); -- Random avatar skin
-            SpawnMob(i):updateEnmity(target);
-            avatar:useMobAbility(912);
-            DespawnMob(i);
-        end
-        PWardenAstralFlows = PWardenAstralFlows + 2;
-    elseif (mobHP <= (mobMaxHP * 0.25) and target:getMaskBit(PWardenAstralFlows,1) == false) then
-        for i=17056178, 17056186 do
-            local rannum = math.random(0,7);
-            local avatar = GetMobByID(i);
-            avatar:changeSkin(23 + rannum); -- Random avatar skin
-            SpawnMob(i):updateEnmity(target);
-            avatar:useMobAbility(912);
-            DespawnMob(i);
+    
+    -- Check for time limit, too
+    if (os.time(t) > depopTime) then
+        for i = 1, 2 do
+            for j = 1, 8 do
+                if pets[i][j]:isSpawned() then
+                    pets[i][j]:despawn();
+                end
+            end
         end
-        PWardenAstralFlows = PWardenAstralFlows + 1;
+        mob:despawn();
```

Comment:
`DespawnMob(mob:getID)` woudl do teh excat same thing. Both of these would have the same flaw: mob randomly getting stuck as per [#1618](https://github.com/DarkstarProject/darkstar/issues/1618#issuecomment-115251959). It doesn't happen everytime but happens enough to be annoying (like the auto homepoint thing) The only way to avoid that I have found is to put the despawn command into a dummy mobskill.